spt/spt-serverplugin.cpp

```diff
@@ -132,22 +132,16 @@ void DefaultFOVChangeCallback(ConVar* var, char const* pOldString)
 	}
 }
 
-//
-// The plugin is a static singleton that is exported as an interface
-//
-#ifdef OE
-CSourcePauseTool g_SourcePauseTool;
-EXPOSE_SINGLE_INTERFACE_GLOBALVAR(CSourcePauseTool,
-                                  IServerPluginCallbacks,
-                                  INTERFACEVERSION_ISERVERPLUGINCALLBACKS_VERSION_1,
-                                  g_SourcePauseTool);
-#else
 CSourcePauseTool g_SourcePauseTool;
-EXPOSE_SINGLE_INTERFACE_GLOBALVAR(CSourcePauseTool,
-                                  IServerPluginCallbacks,
-                                  INTERFACEVERSION_ISERVERPLUGINCALLBACKS,
-                                  g_SourcePauseTool);
-#endif
+
+extern "C" __declspec(dllexport) const void* CreateInterface(const char* pName, int* pReturnCode) {
+	if (pReturnCode) {
+		*pReturnCode = 2007;
```

Comment:
According to the sdk
> if pReturnCode is set, it will return one of the following values (IFACE_OK, IFACE_FAILED)

Also, should we check for the `INTERFACEVERSION_ISERVERPLUGINCALLBACKS`?

Reply:
Checking for the version is not necessary. I also just set the return code to something funny since it isn't actually used.